src/rust/catloop/mod.rs

```diff
@@ -564,38 +612,84 @@ impl CatloopLibOS {
         Err(Fail::new(libc::EINVAL, &cause))
     }
 
+    /// Harvests resources on close.
+    fn harvest_resources_on_close(&mut self, qd: QDesc) -> Result<(), Fail> {
+        // If this was an async_close(), harvest resources.
```

Comment:
How does this function know whether it's called in the `async_close()` path?

Reply:
**Case 1** queue is not associated to a duplex pipe (socket is not bound)
- `async_close()` line 476
- `pack_result()` line 664
- `havest_resources_on_close()`

**Case 2** queue is associated to a duplex pipe (socket is bound)
- `async_close()` line 472
- `pack_result()` line 674
- `havest_resources_on_close()`

Reply:
Oh, I didn't where it was called from. I was referring to the comment. It says that we call it if it was an async close. But there is no check in the function itself.